Logic.json

```diff
@@ -78,16 +78,17 @@
 		"Oogle_Cave_BK": [["Terrydactyland", "Clockwork_Warp"], ["Terrydactyland", "Dinosaur", "TDL_Mumbo"], ["Terrydactyland", "Nest", "Glitches"], ["Terrydactyland", "Split_Up", "SuperBanjo", "Claw_Clambers", "Glitches"], ["Terrydactyland", "Split_Up", "SuperBanjo", "Tricks", "Glitches"]],
 		"Quagmire": [["Wasteland", "Springy_Step_Shoes"]],
 		"Grunty_Industries": [["Quagmire", "JiggyWiggySpecial"], ["Quagmire", "28_Jiggies"]],
-		"Inside_GI": [["Grunty_Industries", "GGM_Mumbo"], ["Grunty_Industries", "Clockwork_Warp"], ["Grunty_Industries", "Clockworks"]],
+		"Inside_GI": [["Grunty_Industries", "GGM_Mumbo"], ["Grunty_Industries", "Clockwork_Warp"], ["Grunty_Industries", "Clockworks", "Glitches"]],
 		"Floor_4_Early": [["Inside_GI", "Explosives", "Egg_Aim", "Glitches"]],
+		"Floor_4_Crushers": [["Roof", "GI_Mumbo", "Washing_Machine", "Bill_Drill"], ["Floor_4_Early"], ["Roof", "Split-Up", "Glitches"]],
 		"Floor_2": [["Inside_GI", "Claw_Clambers"], ["Floor_4_Early"], ["Grunty_Industries", "Clockworks"]],
-		"Roof": [["Floor_2", "Claw_Clambers"], ["Floor_4_Early"], ["Floor_2", "Washing_Machine"], ["Floor_2", "Leg_Spring", "Split_Up"]],
+		"Roof": [["Floor_2", "Claw_Clambers"], ["Floor_4_Early"], ["Floor_2", "Washing_Machine"], ["Floor_2", "Leg_Spring", "Split_Up"], ["Grunty_Industries", "Claw_Clambers", "Tricks"], ["Inside_GI", "Split_Up", "Claw_Clambers", "Leg_Spring", "Wing_Whack"]],
 		"Battery": [["Split_Up", "Pack_Whack", "Taxi_Pack"]],
 		"Toxic_Waste": [["Inside_GI", "Battery"]],
 		"Floor_2_Button": [["Floor_2", "Split_Up", "Leg_Spring"], ["Floor_2", "Claw_Clambers"], ["Roof"]],
-		"Weldar_Door": [["Roof", "GI_Mumbo", "Washing_Machine", "Battery", "Explosives"]],
+		"Weldar_Door": [["Roof", "GI_Mumbo", "Washing_Machine", "Battery", "Grenade_Eggs", "Bill Drill"]],

```

Comment:
I forgot about bill drill youre right. What about the gate between wumba's and the floor 2 battery door? can clockworks blow those up? in which case explosives would be a better fit than grenades.